// react-spotify-player/dist/SpotifyPlayer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// Dimension prop type
var dimensionPropType = _react2.default.PropTypes.oneOfType([_react2.default.PropTypes.number, _react2.default.PropTypes.string]);

// Size presets, defined by Spotify

/**
 * Spotify player iframe widget
 *
 * @author Alexander Wallin <office@alexanderwallin.com>
 * @see https://developer.spotify.com/technologies/widgets/spotify-play-button/
 */

var sizePresets = {
  large: {
    width: 300,
    height: 380
  },
  compact: {
    width: 300,
    height: 80
  }
};

/**
 * SpotifyPlayer class
 */
var SpotifyPlayer = _react2.default.createClass({
  displayName: 'SpotifyPlayer',

  // ------------------------------------------------------
  // Component specs & lifecycle
  // ------------------------------------------------------

  propTypes: {

    // Spotify URI
    uri: _react2.default.PropTypes.string.isRequired,

    // Size as either a preset or as custom dimensions
    size: _react2.default.PropTypes.oneOfType([_react2.default.PropTypes.oneOf(['large', 'compact']), _react2.default.PropTypes.shape({
      width: dimensionPropType,
      height: dimensionPropType
    })]),

    // View
    view: _react2.default.PropTypes.oneOf(['list', 'coverart']),

    // Theme
    theme: _react2.default.PropTypes.oneOf(['black', 'white'])
  },

  getDefaultProps: function getDefaultProps() {
    return {
      size: 'large',
      view: 'list',
      theme: 'black'
    };
  },

  // ------------------------------------------------------
  // Render
  // ------------------------------------------------------

  render: function render() {
    var _props = this.props;
    var uri = _props.uri;
    var view = _props.view;
    var theme = _props.theme;
    var size = this.props.size;

    if (typeof size === 'string') {
      size = sizePresets[size];
    }

    return _react2.default.createElement('iframe', {
      className: 'SpotifyPlayer',
      src: 'https://embed.spotify.com/?uri=' + uri + '&view=' + view + '&theme=' + theme,
      width: size.width,
      height: size.height,
      frameBorder: '0',
      allowTransparency: 'true'
    });
  }
});

exports.default = SpotifyPlayer;
